state-chain/runtime/src/chainflip.rs

```diff
@@ -337,7 +337,7 @@ impl TransactionBuilder<Solana, SolanaApi<SolEnvironment>> for SolanaTransaction
 	fn build_transaction(
 		signed_call: &SolanaApi<SolEnvironment>,
 	) -> <Solana as Chain>::Transaction {
-		signed_call.transaction.clone()
+		SolanaTransactionData { serialized_transaction: signed_call.chain_encoded() }
```

Comment:
In `chain_encoded`, instead of unwrap_or_default, can we also `log_or_panic` so that we catch any errors?